### PR TITLE
CONSOLE-3600: Filter operators based on nodes OS type

### PR DESCRIPTION
**Pull request by jhadvig**
Console-operator changes [CONSOLE-3279](https://issues.redhat.com/browse/CONSOLE-3279) which need to get in first.

Adding screen to confirm that the `console-config.yaml` contains the new field:
Screenshot 2023-04-11 at 12 38 52

/assign @rhamilto @TheRealJon

### CODE REVIEW

**Comment by rhamilto** [comment]
Backend code???  *Looks at @TheRealJon*

**Comment by jhadvig** [comment]
Hold till [CONSOLE-3279](https://issues.redhat.com/browse/CONSOLE-3279) will get merged.

/hold

**Comment by XiyunZhao** [comment]
This PR has been tested by using cluster-bot, now the nodeOperatingSystems would be shown on the console-config.yaml both in Console and CLI (oc describe configmaps console-config -n openshift-console)
/lgtm
/label qe-approved

File: cmd/bridge/main.go
Comment (TheRealJon) [inline cmd/bridge/main.go:273]: nit, we should eventually implement a custom ListFlag for these.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:56]: nit, prefer to remove lodash if we can.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:88]: nit, prefer to remove lodash if we can

**Review by TheRealJon — APPROVED** [review]
Just a couple of nits, but I think it's good to merge.
/lgtm

**Comment by jhadvig** [comment]
QE Approver:
 /assign @XiyunZhao 
Docs Approver:
 /assign @opayne1
PX Approver:
/assign @RickJWagner

**Comment by XiyunZhao** [comment]
Remove the qe-approved label after chatting with the developer
/remove-label qe-approved

**Comment by XiyunZhao** [comment]
This PR has been retested by using cluster-bot, and private operator, now new flag nodeOperatingSystems has been added to the OCP successfully, and the filtering logic in the operator hub is updated
/label qe-approved

**Pull request merged**
